doc/tutorial/network_ring.rst

```diff
@@ -0,0 +1,261 @@
+      def connections_on(self, gid):
+         src = (gid-1)%self.ncells
+         w = 0.01
+         d = 5
+         return [arbor.connection((src,0), (gid,0), w, d)]
+
+      def num_targets(self, gid):
+         return 1
+
+      def num_sources(self, gid):
+         return 1
+
+      # (9) Attach a generator to the first cell in the ring.
+      def event_generators(self, gid):
+         if gid==0:
+               sched = arbor.explicit_schedule([1])
+               return [arbor.event_generator((0,0), 0.1, sched)]
+         return []
+
+      # (10) Place a probe at the root of each cell.
+      def probes(self, gid):
+         return [arbor.cable_probe_membrane_voltage('"root"')]
+
+      def global_properties(self, kind):
+         return self.props
+
+   # (11) Instantiate recipe
+   ncells = 4
+   recipe = ring_recipe(ncells)
+
+Run the simulation
+******************
+
+To create a simulation, we must create an :class:`arbor.context` and :py:class:`arbor.domain_decomposition`.
+
+Step **(12)** creates a default execution context, and uses the :func:`arbor.partition_load_balance` to create a
+default domain decomposition. You can print the objects to see what defaults they produce on your system.
+
+Step **(13)** sets all spike generators to record using the :py:class:`arbor.spike_recording.all` policy.
+This means the timestamps of the generated events will be kept in memory. Be default, these are discarded.
+
+In addition to having the timestamps of spikes, we want to extract the voltage as a function of time.
+
+Step **(14)** sets the probes (step **10**) to measure at a certain schedule. This is sometimes described as attaching a :term:`sampler` to a :term:`probe`. :py:func:`arbor.simulation.sample` expects a :term:`probe id` and the desired schedule (here: a recording frequency of 10 kHz). Note that the probe id is a separate index from those of :term:`connection` endpoints; probe ids correspond to the index of the list produced by :py:func:`arbor.recipe.probes` on cell ``gid``.
+
+:py:func:`arbor.simulation.sample` returns a handle to the :term`samples <sample>` that will be recorded. We store these handles for later use.
+
+Step **(15)** executes the simulation for a duration of 100 ms.
+
+.. code-block:: python
+
+   # (12) Create a default execution context, domain decomposition and simulation
+   context = arbor.context()
+   decomp = arbor.partition_load_balance(recipe, context)
+   sim = arbor.simulation(recipe, decomp, context)
+
+   # (13) Set spike generators to record
+   sim.record(arbor.spike_recording.all)
+
+   # (14) Attach a sampler to the voltage probe on cell 0. Sample rate of 10 sample every ms.
+   handles = [sim.sample((gid, 0), arbor.regular_schedule(0.1)) for gid in range(ncells)]
+
+   # (15) Run simulation
+   sim.run(100)
+   print('Simulation finished')
+
+The results
+***********
+
+Step **(16)**: we can print the timestamps of the spikes:
```

Comment:
Minor nitpick. For the previous steps, you used `step x does/creates/starts` etc. Can we continue the same patter for steps 16 and 17?